actions: less warnings is more (fixes #3671)

ewhang5 opened the pull request:
fixes #3671

Okuro3499 commented on app/src/main/java/org/ole/planet/myplanet/ui/sync/LoginActivity.kt:
```diff
@@ -60,8 +61,8 @@ class LoginActivity : SyncActivity(), TeamListAdapter.OnItemClickListener {
         }
-        if (intent.hasExtra("versionInfo")) {
-            onUpdateAvailable((intent.getSerializableExtra("versionInfo") as MyPlanet?),
+        if (intent.hasExtra("versionInfo") && Build.VERSION.SDK_INT >= TIRAMISU) {
+            onUpdateAvailable(intent.getSerializableExtra("versionInfo", MyPlanet::class.java),
```
consider also older android versions I don't think it will handle this change. refer [here](https://github.com/open-learning-exchange/myplanet/pull/3634/files#diff-b43e9f15f676f3708da7dbc49a0c8cfa0384d6d3110df3d77e5c22364cac1767R82-R87) on how it was done for something similar

ewhang5 commented:
@Okuro3499 addressed your comment.

Okuro3499 reviewed (changes requested):
@ewhang5 this can be refactored to something like this. currently it seems to have repetitive lines

```
val versionInfo = if (Build.VERSION.SDK_INT >= TIRAMISU) {
    intent.getSerializableExtra("versionInfo", MyPlanet::class.java)
} else {
    @Suppress("DEPRECATION")
    intent.getSerializableExtra("versionInfo") as? MyPlanet
}

if (versionInfo != null) {
    onUpdateAvailable(versionInfo, intent.getBooleanExtra("cancelable", false))
} else {
    service.checkVersion(this, settings)
}
```

ewhang5 commented:
> @ewhang5 this can be refactored to something like this. currently it seems to have repetitive lines
> 
> ```
> val versionInfo = if (Build.VERSION.SDK_INT >= TIRAMISU) {
>     intent.getSerializableExtra("versionInfo", MyPlanet::class.java)
> } else {
>     @Suppress("DEPRECATION")
>     intent.getSerializableExtra("versionInfo") as? MyPlanet
> }
> 
> if (versionInfo != null) {
>     onUpdateAvailable(versionInfo, intent.getBooleanExtra("cancelable", false))
> } else {
>     service.checkVersion(this, settings)
> }
> ```

addressed your comment
